### PR TITLE
Adjust Random Beacon DKG after the changes in ECDSA DKG

### PR DESCRIPTION
**Pull request by nkuba**
Refs: https://github.com/keep-network/keep-core/issues/2816

This PR implements changes introduced in the ECDSA project that are common to both projects.

- [x] emit the entire DKG result in DkgResultSubmitted
- [x] register the group only when DKG result was accepted
- [x] wrap calls to TokenStaking.seize in a try-catch when challenging the result,
- [x] prefix duplicated contract names (DKG -> BeaconDkg, DKGValidator -> BeaconDkgValidator)
- [x] ensure consistent code formatting with ecdsa project

This PR doesn't cover reimbursements in ETH instead of T token.

### CODE REVIEW

**Comment by pdyraga** [comment]
Let's also address https://github.com/keep-network/keep-core/pull/2801#discussion_r809961088

**Comment by nkuba** [comment]
> Let's also address [#2801 (comment)](https://github.com/keep-network/keep-core/pull/2801#discussion_r809961088)

43e6b44194ccff1417b01a216b0fbacea7d13b4c

File: solidity/random-beacon/test/RandomBeacon.GroupCreation.test.ts [hunk 1]
```diff
@@ -2771,18 +2736,25 @@ describe("RandomBeacon - Group Creation", () => {
...
 
   expect(
     dkgData.parameters.resultChallengePeriodLength,
     "unexpected resultChallengePeriodLength"
   ).to.eq(params.dkgResultChallengePeriodLength)
 
   expect(
-    dkgData.parameters.resultSubmissionEligibilityDelay,
+    dkgData.parameters.resultSubmissionTimeout,
     "unexpected resultSubmissionEligibilityDelay"
```
Comment (dimpar) [inline solidity/random-beacon/test/RandomBeacon.GroupCreation.test.ts:2751]: resultSubmissionEligibilityDelay -> resultSubmissionTimeout

**Comment by dimpar** [comment]
Do we know if all these changes that were done post-audit to `/random-beacon` will be audited again? If the answer is yes, how about we also change the [earlier discussed](https://github.com/keep-network/keep-core/pull/2772#discussion_r776715819) `getSlashingFactor()` to simplify slashing calculation and cut some gas?

File: solidity/random-beacon/test/RandomBeacon.GroupCreation.test.ts [hunk 2]
```diff
@@ -63,33 +74,32 @@ const fixture = async () => {
...
   let randomBeaconGovernance: RandomBeaconGovernance
-  let randomBeacon: RandomBeaconStub & RandomBeacon
+  let randomBeacon: RandomBeaconTest
   let sortitionPool: SortitionPool
   let staking: StakingStub
   let testToken: TestToken
 
   before(async () => {
     thirdParty = await ethers.getSigner((await getUnnamedAccounts())[0])
+    let randomBeaconStub
```
Comment (dimpar) [inline solidity/random-beacon/test/RandomBeacon.GroupCreation.test.ts:91]: how about
```
let randomBeaconStub: RandomBeaconTest
(...)
randomBeacon = randomBeaconStub
```

File: solidity/random-beacon/test/RandomBeacon.GroupCreation.test.ts [hunk 3]
```diff
@@ -127,7 +137,7 @@ describe("RandomBeacon - Group Creation", () => {
       })
 
       it("should lock the sortition pool", async () => {
-        expect(await sortitionPool.isLocked()).to.be.true
+        await expect(await sortitionPool.isLocked()).to.be.true
```
Comment (dimpar) [inline solidity/random-beacon/test/RandomBeacon.GroupCreation.test.ts:140]: we can remove "await" from `await expect(...)`

File: solidity/random-beacon/contracts/RandomBeacon.sol [hunk 4]
```diff
@@ -373,27 +379,25 @@ contract RandomBeacon is IRandomBeacon, Ownable {
     ///      random beacon governance contract. The caller is responsible for
     ///      validating parameters.
     /// @param _relayRequestFee New relay request fee
-    /// @param _relayEntrySubmissionEligibilityDelay New relay entry submission
+    /// @param _relayEntrySoftTimeout New relay entry submission
```
Comment (dimpar) [inline solidity/random-beacon/contracts/RandomBeacon.sol:382]: New relay entry submission soft timout

File: solidity/random-beacon/contracts/RandomBeaconGovernance.sol [hunk 5]
```diff
@@ -568,56 +572,101 @@ contract RandomBeaconGovernance is Ownable {
...
         dkgResultChallengePeriodLengthChangeInitiated = 0;
         newDkgResultChallengePeriodLength = 0;
     }
 
-    /// @notice Begins the DKG result submission eligibility delay update
+    /// @notice Begins the DKG result submission timeout update
     ///         process.
     /// @dev Can be called only by the contract owner.
-    /// @param _newDkgResultSubmissionEligibilityDelay New DKG result submission
+    /// @param _newDkgResultSubmissionTimeout New DKG result submission
```
Comment (dimpar) [inline solidity/random-beacon/contracts/RandomBeaconGovernance.sol:585]: ..result submission timeout..

We have 2 more places in this test if you search for "eligibility delay"

File: solidity/random-beacon/contracts/libraries/BeaconDkg.sol [hunk 6]
```diff
@@ -220,7 +217,7 @@ library DKG {
         emit DkgStarted(seed);
```
[the remark this answers is not in the document]
Comment (nkuba) [reply solidity/random-beacon/contracts/libraries/BeaconDkg.sol]: 841e683779e762dee6696e0fb1e53c19d0e257be

File: solidity/random-beacon/contracts/RandomBeacon.sol [hunk 7]
```diff
@@ -335,11 +340,12 @@ contract RandomBeacon is IRandomBeacon, Ownable {
 
         dkg.init(_sortitionPool, _dkgValidator);
         dkg.setResultChallengePeriodLength(11520); // ~48h assuming 15s block time
-        dkg.setResultSubmissionEligibilityDelay(20);
+        dkg.setResultSubmissionTimeout(64 * 20); // TODO: Verify value
```
[the remark this answers is not in the document]
Comment (nkuba) [reply solidity/random-beacon/contracts/RandomBeacon.sol]: 27c9c40b22dc486d3f8f41644fff26560cb614b0

File: solidity/random-beacon/contracts/RandomBeacon.sol [hunk 8]
```diff
@@ -801,18 +816,29 @@ contract RandomBeacon is IRandomBeacon, Ownable {
...
-        );
-
-        staking.seize(
-            slashingAmount,
-            relayEntryTimeoutNotificationRewardMultiplier,
-            msg.sender,
-            groupMembersAddresses
-        );
+        try
+            staking.seize(
```
[the remark this answers is not in the document]
Comment (nkuba) [reply solidity/random-beacon/contracts/RandomBeacon.sol:826]: a614f949be875c419389582829bd545fc5967cce

File: solidity/random-beacon/test/RandomBeacon.GroupCreation.test.ts (hunk 1 above)
Comment (nkuba) [reply solidity/random-beacon/test/RandomBeacon.GroupCreation.test.ts]: 9ad4a491c75fe210a41b1235dcbc49f6b15938d3

**Comment by nkuba** [comment]
> 

Let's try!

**Comment by nkuba** [comment]
> Do we know if all these changes that were done post-audit to `/random-beacon` will be audited again? If the answer is yes, how about we also change the [earlier discussed](https://github.com/keep-network/keep-core/pull/2772#discussion_r776715819) `getSlashingFactor()` to simplify slashing calculation and cut some gas? I can take care of it in a separate PR if we all agree.

Let's try!

File: solidity/random-beacon/contracts/RandomBeacon.sol (hunk 4 above)
Comment (nkuba) [reply solidity/random-beacon/contracts/RandomBeacon.sol]: 6ae27026d2c534e5b17b254c048cdd77403b95f4

File: solidity/random-beacon/contracts/RandomBeaconGovernance.sol (hunk 5 above)
Comment (nkuba) [reply solidity/random-beacon/contracts/RandomBeaconGovernance.sol:585]: 6ae27026d2c534e5b17b254c048cdd77403b95f4

File: solidity/random-beacon/test/RandomBeacon.GroupCreation.test.ts (hunk 3 above)
Comment (nkuba) [reply solidity/random-beacon/test/RandomBeacon.GroupCreation.test.ts]: 6e5ced1cc2d78054a88226d3ea45fd8f605c2bd1

File: solidity/random-beacon/test/RandomBeacon.GroupCreation.test.ts (hunk 2 above)
Comment (nkuba) [reply solidity/random-beacon/test/RandomBeacon.GroupCreation.test.ts]: 6e5ced1cc2d78054a88226d3ea45fd8f605c2bd1

File: solidity/random-beacon/contracts/libraries/Relay.sol [hunk 9]
```diff
@@ -189,17 +181,15 @@ library Relay {
...
         Data storage self,
-        uint256 newRelayEntrySubmissionEligibilityDelay
+        uint256 newRelayEntrySoftTimeout
     ) internal {
         require(!isRequestInProgress(self), "Relay request in progress");
 
-        self.relayEntrySubmissionEligibilityDelay = uint32(
-            newRelayEntrySubmissionEligibilityDelay
-        );
+        self.relayEntrySoftTimeout = uint32(newRelayEntrySoftTimeout);
```
Comment (dimpar) [inline solidity/random-beacon/contracts/libraries/Relay.sol:192]: Thoughts on removing casting by either defining soft/hard timeouts as `uint256` or passing `uint32`?

**Review by dimpar — COMMENTED** [review]
No more comments from my side.

File: solidity/random-beacon/test/system/e2e.test.ts [hunk 10]
```diff
@@ -138,7 +140,6 @@ describe("System -- e2e", () => {
         .approveDkgResult(dkgResult.dkgResult)
 
       // Ids of groups that sign a given request in order
```
[the remark this answers is not in the document]
Comment (nkuba) [reply solidity/random-beacon/test/system/e2e.test.ts]: 0af5b206c4acb661e04a3481ae2fd5577a0d0d77

File: solidity/random-beacon/contracts/libraries/Relay.sol (hunk 9 above)
Comment (nkuba) [reply solidity/random-beacon/contracts/libraries/Relay.sol:192]: I'd leave it as is in this PR for consistency. We should consider updating it in a separate PR to determine how changes will impact gas costs.

**Comment by nkuba** [comment]
### BEFORE

```
·--------------------------------------------------------------------------------------------|---------------------------|-------------|-----------------------------·
|                                    Solc version: 0.8.9                                     ·  Optimizer enabled: true  ·  Runs: 200  ·  Block limit: 30000000 gas  │
·····························································································|···························|·············|······························
|  Methods                                                                                                                                                           │
···························|·································································|·············|·············|·············|···············|··············
|  Contract                ·  Method                                                         ·  Min        ·  Max        ·  Avg        ·  # calls      ·  eur (avg)  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  approveDkgResult                                               ·     186381  ·     247866  ·     206072  ·           26  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  challengeDkgResult                                             ·     334131  ·    1285713  ·    1008153  ·           26  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  fundDkgRewardsPool                                             ·      44305  ·      78505  ·      61405  ·            2  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  fundHeartbeatNotifierRewardsPool                               ·      78528  ·      78540  ·      78533  ·            5  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  notifyDkgTimeout                                               ·      85216  ·      87216  ·      86549  ·            3  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  notifyFailedHeartbeat                                          ·     631677  ·     885717  ·     767132  ·           12  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  publicDkgLockState                                             ·          -  ·          -  ·      33643  ·            2  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  registerOperator                                               ·     205994  ·     342733  ·     211901  ·          259  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  reportRelayEntryTimeout                                        ·     405745  ·     418678  ·     412540  ·           14  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  reportUnauthorizedSigning                                      ·          -  ·          -  ·     527877  ·            3  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  requestRelayEntry                                              ·      78905  ·     152345  ·     115194  ·           32  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  roughlyAddGroup                                                ·          -  ·          -  ·     119227  ·            2  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  roughlyTerminateGroup                                          ·          -  ·          -  ·      72617  ·            2  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  submitDkgResult                                                ·     297260  ·     421650  ·     382716  ·           66  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  submitRelayEntry                                               ·     192140  ·     241767  ·     209021  ·           26  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  submitRelayEntry                                               ·     214192  ·     495285  ·     354739  ·            8  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  Deployments                                                                               ·                                         ·  % of limit   ·             │
·····························································································|·············|·············|·············|···············|··············
|  BLS                                                                                       ·          -  ·          -  ·     714061  ·        2.4 %  ·          -  │
·····························································································|·············|·············|·············|···············|··············
|  CallbackContractStub                                                                      ·          -  ·          -  ·     142873  ·        0.5 %  ·          -  │
·····························································································|·············|·············|·············|···············|··············
|  DKG                                                                                       ·          -  ·          -  ·    1286376  ·        4.3 %  ·          -  │
·····························································································|·············|·············|·············|···············|··············
|  DKGValidator                                                                              ·    1429083  ·    1429095  ·    1429093  ·        4.8 %  ·          -  │
·····························································································|·············|·············|·············|···············|··············
|  GroupsStub                                                                                ·          -  ·          -  ·    1167526  ·        3.9 %  ·          -  │
·····························································································|·············|·············|·············|···············|··············
|  Heartbeat                                                                                 ·          -  ·          -  ·     980109  ·        3.3 %  ·          -  │
·····························································································|·············|·············|·············|···············|··············
|  RandomBeaconGovernance                                                                    ·          -  ·          -  ·    4830871  ·       16.1 %  ·          -  │
·····························································································|·············|·············|·············|···············|··············
|  RandomBeaconStub                                                                          ·    5642663  ·    5642687  ·    5642687  ·       18.8 %  ·          -  │
·····························································································|·············|·············|·············|···············|··············
|  RelayStub                                                                                 ·          -  ·          -  ·     256172  ·        0.9 %  ·          -  │
·····························································································|·············|·············|·············|···············|··············
|  SortitionPool                                                                             ·          -  ·          -  ·    2245461  ·        7.5 %  ·          -  │
·····························································································|·············|·············|·············|···············|··············
|  StakingStub                                                                               ·          -  ·          -  ·     319203  ·        1.1 %  ·          -  │
·····························································································|·············|·············|·············|···············|··············
|  SubmissionStub                                                                            ·          -  ·          -  ·     197161  ·        0.7 %  ·          -  │
·····························································································|·············|·············|·············|···············|··············
|  TestAltBn128                                                                              ·          -  ·          -  ·    3693821  ·       12.3 %  ·          -  │
·····························································································|·············|·············|·············|···············|··············
|  TestModUtils                                                                              ·          -  ·          -  ·    5742491  ·       19.1 %  ·          -  │
·····························································································|·············|·············|·············|···············|··············
|  TestToken                                                                                 ·          -  ·          -  ·    1271349  ·        4.2 %  ·          -  │
·--------------------------------------------------------------------------------------------|-------------|-------------|-------------|---------------|-------------·
```

### AFTER

```
·--------------------------------------------------------------------------------------------|---------------------------|-------------|-----------------------------·
|                                    Solc version: 0.8.9                                     ·  Optimizer enabled: true  ·  Runs: 200  ·  Block limit: 30000000 gas  │
·····························································································|···························|·············|······························
|  Methods                                                                                                                                                           │
···························|·································································|·············|·············|·············|···············|··············
|  Contract                ·  Method                                                         ·  Min        ·  Max        ·  Avg        ·  # calls      ·  eur (avg)  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  approveDkgResult                                               ·     349084  ·     436680  ·     373659  ·           27  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  challengeDkgResult                                             ·     316917  ·    1265550  ·    1015708  ·           26  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  genesis                                                        ·          -  ·          -  ·      82630  ·           22  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  notifyDkgTimeout                                               ·      85118  ·      87118  ·      86618  ·            4  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  notifyFailedHeartbeat                                          ·     631677  ·     885717  ·     767132  ·           12  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  publicDkgLockState                                             ·          -  ·          -  ·      33643  ·            2  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  registerOperator                                               ·     206060  ·     342799  ·     211967  ·          259  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  reportRelayEntryTimeout                                        ·     369381  ·     413110  ·     404866  ·           16  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  reportUnauthorizedSigning                                      ·     492250  ·     492255  ·     492252  ·            5  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  requestRelayEntry                                              ·      77577  ·     151005  ·     112739  ·           32  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  roughlyAddGroup                                                ·          -  ·          -  ·     119271  ·            2  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  roughlyTerminateGroup                                          ·          -  ·          -  ·      72572  ·            2  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  submitDkgResult                                                ·     232415  ·     239879  ·     235142  ·           68  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  submitRelayEntry                                               ·     192070  ·     241697  ·     208950  ·           26  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  submitRelayEntry                                               ·     214122  ·     495215  ·     354669  ·            8  ·          -  │
···························|·································································|·············|·············|·············|···············|··············
|  Deployments                                                                               ·                                         ·  % of limit   ·             │
·····························································································|·············|·············|·············|···············|··············
|  BeaconDkg                                                                                 ·          -  ·          -  ·    1116089  ·        3.7 %  ·          -  │
·····························································································|·············|·············|·············|···············|··············
|  BeaconDkgValidator                                                                        ·          -  ·          -  ·    1429095  ·        4.8 %  ·          -  │
·····························································································|·············|·············|·············|···············|··············
|  BLS                                                                                       ·          -  ·          -  ·     714061  ·        2.4 %  ·          -  │
·····························································································|·············|·············|·············|···············|··············
|  CallbackContractStub                                                                      ·          -  ·          -  ·     142873  ·        0.5 %  ·          -  │
·····························································································|·············|·············|·············|···············|··············
|  GroupsStub                                                                                ·          -  ·          -  ·     998833  ·        3.3 %  ·          -  │
·····························································································|·············|·············|·············|···············|··············
|  Heartbeat                                                                                 ·          -  ·          -  ·     980109  ·        3.3 %  ·          -  │
·····························································································|·············|·············|·············|···············|··············
|  RandomBeaconGovernance                                                                    ·          -  ·          -  ·    5097303  ·         17 %  ·          -  │
·····························································································|·············|·············|·············|···············|··············
|  RandomBeaconStub                                                                          ·    5615892  ·    5615904  ·    5615901  ·       18.7 %  ·          -  │
·····························································································|·············|·············|·············|···············|··············
|  RelayStub                                                                                 ·          -  ·          -  ·     250058  ·        0.8 %  ·          -  │
·····························································································|·············|·············|·············|···············|··············
|  SortitionPool                                                                             ·          -  ·          -  ·    2245461  ·        7.5 %  ·          -  │
·····························································································|·············|·············|·············|···············|··············
|  StakingStub                                                                               ·          -  ·          -  ·     319203  ·        1.1 %  ·          -  │
·····························································································|·············|·············|·············|···············|··············
|  TestAltBn128                                                                              ·          -  ·          -  ·    3693821  ·       12.3 %  ·          -  │
·····························································································|·············|·············|·············|···············|··············
|  TestModUtils                                                                              ·          -  ·          -  ·    5742491  ·       19.1 %  ·          -  │
·····························································································|·············|·············|·············|···············|··············
|  TestToken                                                                                 ·          -  ·          -  ·    1271349  ·        4.2 %  ·          -  │
·--------------------------------------------------------------------------------------------|-------------|-------------|-------------|---------------|-------------·
```

**Review by dimpar — APPROVED** [review]
@pdyraga wanna take a look?

**Pull request merged**